proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+		<param name="weatherImageName" type="String" mandatory="false">
+	</struct>
+    
+```
+
+
+#### Navigation
+A navigation service is defined as a service that is currently listed as the navigation provider. 
+
+```xml
+	<struct name="NavigationServiceManifest">
+		<param name="acceptsWayPoints" type="Boolean" mandatory="false"/>
+	</struct>
+
+	<struct name="NavigationServiceData">
+		<description> This data is related to what a navigation service would provide</description>
+		
+		<param name="waypoints" type="LocationDetails" array="true" minsize="2" mandatory="false">
+			<description> This array should be ordered with the first object being the start location of the current trip and the last object the final destination. </description>
+		</param>
+		
+		<param name="eta" type="DateTime" mandatory="false"/>
+
+	</struct>
+	
+    
+```
+
+###### RPCs to be handled:
+ - SendLocation
+ - GetWayPoints
+
+
+#### Voice Assistant
+A voice assistant service is defined as a service that is currently acting as the voice assistant provider. This provider needs to be flushed out with a lot of extra details as it's functionality is far reaching.
+
+```xml
+
+	<struct name="VoiceAssistantServiceManifest">
+		<param name="wakeWords" type="String" array="true" minsize="1" mandatory="false"/>
+		
+	</struct>
+	
+	<struct name="VoiceAssistantServiceData">
+		<description> This data is related to what a voice assistant service would provide</description>
+
+	</struct>
+
+	<enum name="VoiceAssistantTrigger">
+		<element name="PUSH_TO_TALK_BUTTON"/>
+		<element name="WAKE_WORD"/>
+	</enum>
+```
+
+##### New RPCs for Voice Assistant
+
+```xml
+	<function name="OnVoiceAssistantActivated" functionID="OnVoiceAssistantActivatedID" messagetype="notification">
+		<param name="triggerSource" type="VoiceAssistantTrigger" mandatory="true">
+			<description>Informs the voice assistant of which source triggered the event.</description>
+		</param>
+		<param name="triggerInfo" type="String" mandatory="false">
+			<description>Any extra information about the trigger. This should include the wake word used and any other speech strings recognized.</description>
+		</param>
+	
+	</function>
+	
+	<struct name="VRSynonym">
+		<param name="synonym" type="String" mandatory="true"/>
+		<param name="id" type="Integer" mandatory="true"/>
+	</struct>
+	
+	<function name="UpdateVRSynonyms" functionID="UpdateVRSynonymsID" messagetype="request">
+			<param name="vrSynonyms" type="VRSynonym" array="true" mandatory="true"/>
+			<param name="shouldAdd" type="Boolean" mandatory="true">
+				<description>Set to true if the synonyms should be added. Set to false if they should be removed</description>
+			</param>
+	</function>
+	
+	
+	<function name="UpdateVRSynonyms" functionID="UpdateVRSynonymsID" messagetype="response">
+		<param name="success" type="Boolean" platform="documentation" mandatory="true">
+			<description> true if successful; false, if failed </description>
```

Comment:
Should there be a comma after "true" or should the comma after "false" be removed so that these match?